File: bindings/csharp/FastParse/FastParseNormalization.cs
namespace FastParse;

/// <summary>
/// Source normalization modes applied in native memory before Tree-sitter parses the source.
/// </summary>
public enum FastParseNormalization
{
    /// <summary>Apply safe language-specific legacy cleanup when FastParse knows it is appropriate.</summary>
    AutoSafe = 0,

    /// <summary>Do not alter source bytes before parsing.</summary>
    None = 1,

    /// <summary>Apply COBOL fixed-format legacy cleanup and conservative parser-view repairs.</summary>
    CobolFixedLegacy = 2
}

File: bindings/java/fastparse/src/main/java/dev/fastparse/FastParseNormalization.java
package dev.fastparse;

/** Source normalization mode applied before Tree-sitter parsing. */
public enum FastParseNormalization {
    AUTO_SAFE(0),
    NONE(1),
    COBOL_FIXED_LEGACY(2);

    private final int nativeValue;

    FastParseNormalization(int nativeValue) {
        this.nativeValue = nativeValue;
    }

    public int nativeValue() {
        return nativeValue;
    }
}
